<launch>
    <arg name="image" default="image_raw"/>
    <arg name="camera" default="camera_info"/>
    <arg name="calibration" default="$(find uvc_cam)/calib/openMV_pinhole.yaml"/>
    <arg name="rectify" default="true"/>
    <arg name="cam_name" default="camera"/>

    <group ns="$(arg cam_name)">

	<!-- camera node, set parameters in uvc_cam_node.py -->
        <node pkg="uvc_cam" name="uvc_cam" type="uvc_cam_node.py" output="screen">
            <param name="image" value="$(arg image)"/>
            <param name="camera" value="$(arg camera)"/>
            <param name="calibration" value="$(arg calibration)"/>
            <rosparam file="$(find uvc_cam)/config/$(arg cam_name).yaml"/>
        </node>

	<!-- image rectification node -->
        <node pkg="image_proc" name="image_proc" type="image_proc" if="$(arg rectify)"/>

    </group>
</launch>
